README.md

```diff
@@ -76,7 +76,16 @@ Feed::Item.desc(:position).limit(5).scroll do |record, next_cursor|
 end
 ```
 
-Resume iterating using the previously saved cursor.
+Resume iterating using saved cursor and save the cursor to go backward.
+
+```ruby
+Feed::Item.desc(:position).limit(5).scroll(saved_cursor) do |record, _, previous_cursor|
+  # each record, one-by-one
+  saved_cursor = previous_cursor
+end
```

Comment:
We now yield the next and previous cursor every time, so everywhere we say `Feed::Item.desc(:position).limit(5).scroll do |record, next_cursor|` it should be `Feed::Item.desc(:position).limit(5).scroll do |record, next_cursor, previous_cursor|`.

I'd rename `saved_cursor` to `saved_next_cursor` and `saved_previous_cursor` to make it a bit clearer.

Reply:
WDYT about the fact that `next_cursor` is before `previous_cursor` ? If we can break things, I suggest having `cursors` which is a hash or object with different cursors so it will give : 

```
Feed::Item.desc(:position).limit(5).scroll(saved_cursor) do |record, cursors|
  # each record, one-by-one
  saved_cursors = cursors[:next]
end

saved_cursors[:next] # => next cursor
saved_cursors[:previous] # => previous cursor 
``` 

IMO, this is better, but it's not backward compatible.

Reply:
A backwards compatible version could be `|record, next_cursor, iterator|` where `iterator` is a class that has `.next_cursor` and `.previous_cursor` and whatever else we choose to add in the future.

It's kinda ugly, so I would also be OK with breaking backwards compatibility, but I do think we should return an object and not just a hash of cursors.

1. Return an object instead of a cursors hash.
2. Increment version to 2.0 in this PR.
3. Add a section to https://github.com/mongoid/mongoid-scroll/blob/master/UPGRADING.md.

Reply:
I did changes.